torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -924,9 +924,7 @@ def redistribute(
     new_spec = copy.deepcopy(old_spec)
     new_spec.placements = placements
     if shard_order is not None:
-        new_spec.shard_order = shard_order
-    else:
-        new_spec.shard_order = ()
+        assert new_spec._maybe_update_placements_given_shard_order(shard_order)
```

Comment:
what happens if placements contain StridedShard and shard_order is also specified? should we assert this can't happen?